lib/membrane_http_adaptive_stream/sink.ex

```diff
@@ -271,8 +271,10 @@ defmodule Membrane.HTTPAdaptiveStream.Sink do
 
   @impl true
   def handle_end_of_stream(Pad.ref(:input, id), _ctx, state) do
-    %{manifest: manifest, storage: storage} = state
+    {storage, manifest} = maybe_finalize_and_store_segment(id, state)
```

Comment:
I'd put the contents of this function here

Reply:
Ok, the main problem is that the name `maybe_finalize_and_store_segment` doesn't suggest it's only valid for end_of_stream. It took me a while to realize why `handle_partial_segment` doesn't use it.

Reply:
I don't want to do that java styles but wdyt about `maybe_finalize_and_store_segment_on_eos(...)`? (`eos` so the name doesn't become a poem).